src/sentry/integrations/vercel/webhook.py

```diff
@@ -44,4 +54,153 @@ def post(self, request):
+                    )
+                    sentry_app_installation = installation_for_provider.sentry_app_installation
+                except SentryAppInstallationForProvider.DoesNotExist:
+                    logger.info("Installation not found", extra=logging_params)
+                    return self.respond({"detail": "Installation not found"}, status=404)
+
+                logging_params["sentry_app_installation_id"] = sentry_app_installation.id
+
+                # find a token associated with the installation so we can use it for authentication
+                try:
+                    sentry_app_installation_token = (
+                        SentryAppInstallationToken.objects.select_related("api_token")
+                        .filter(sentry_app_installation=sentry_app_installation)
+                        .first()
+                    )
+                except SentryAppInstallationToken.DoesNotExist:
+                    logger.info("Token not found", extra=logging_params)
+                    return self.respond({"detail": "Token not found"}, status=404)
+
+                # find the commmit sha so we can  use it as as the release
+                commit_sha = (
+                    meta.get("githubCommitSha")
+                    or meta.get("gitlabCommitSha")
+                    or meta.get("bitbucketCommitSha")
+                )
+
+                # contruct the repo depeding what provider we use
+                if meta.get("githubCommitSha"):
+                    # There is also githubRepo and githubRepo
+                    repository = u"%s/%s" % (meta["githubCommitOrg"], meta["githubCommitRepo"])
+                elif meta.get("gitlabCommitSha"):
+                    # gitlab repos are formatted with a space for some reason
+                    repository = u"%s / %s" % (
+                        meta["gitlabProjectNamespace"],
+                        meta["gitlabProjectName"],
+                    )
+                elif meta.get("bitbucketCommitSha"):
+                    repository = u"%s/%s" % (meta["bitbucketRepoOwner"], meta["bitbucketRepoName"])
+                else:
+                    logger.info("No commit sha", extra=logging_params)
+                    return self.respond({"detail": "No commit sha"}, status=400)
+
+                data = {
+                    "version": commit_sha,
+                    "projects": [project.slug],
+                }
+
+                session = http.build_session()
+                url = absolute_uri("/api/0/organizations/%s/releases/" % organization.slug)
+                headers = {
+                    "Accept": "application/json",
+                    "Authorization": "Bearer %s" % sentry_app_installation_token.api_token.token,
+                }
+                json_error = None
+
+                # create the release
+                try:
+                    resp = session.post(url, json=data, headers=headers,)
```

Comment:
it looks like we use `from sentry.api import client` in the action endpoint for Slack when making calls to sentry, should we using that instead of importing `http`?

Reply:
@MeredithAnya I tried that but we can't pass in a custom header that way. We'd need to provide a user and auth which would require more lookups than what we already have.